### DISCUSSION TITLE
Welcome to Python Like You Mean It Discussions!

### DISCUSSION THREAD

**Discussion by rsokl** [rsokl/Learning_Python#163] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by pyshy** [comment]
Hi! I have experience in other programming languages but I am new to the Business Analytics world and I am new to python as well and would like to learn not just python but the zen of python!

**Reply by rsokl** [reply]
Welcome! Now that you mention it, I realize that I should discuss the zen of Python somewhere in PLYMI!

**Reply by shekhar-banerjee96** [reply]
Hi !! Everyone

**Reply by rsokl** [reply]
Hi @Banerjee1996 ! Welcome!

**Comment by ramezwy5** [comment]
Hi everyone,
I just start exploring python this week with no programming background. my previous experience is computer hardware repair and OS troubleshooting.
I got here while searching for resources about python. which is my good luck.
hope to add good things here

**Reply by rsokl** [reply]
Welcome, @ramezwy5 ! It's really cool that you are coming here from a hardware background; I've always wished that I knew more about the hardware side of things. 

I hope that you find this resource useful!

**Comment by yatinyatin** [comment]
Hello everyone! I am a new programming student with no experience prior. I am having a lot of trouble finding where to complete coursework as I don't know if I should be using Anaconda or GitHub. Can someone please let me know where to complete the coursework? I am currently on the question regarding Jupyter notebooks and their contents. Can you please direct me to where I can find the solutions to this problem?

**Reply by ArtichokeSap** [reply]
@yatinyatin , are you talking about Python Like You Mean It (PLYMI) as a prerequisite for something else, like the Beaver Works Summer Institute?  The PLYMI site/project/Github is standalone without "coursework" per se.  If we knew what you were referring to, we could maybe point you in the right direction.

**Comment by bahamas2020** [comment]
Good Morning :
Nice to hear from you
Lahcene

**Reply by rsokl** [reply]
Hello! I am sorry for missing this earlier. Welcome :)

**Comment by Ussamashaukat** [comment]
Hello

**Reply by rsokl** [reply]
Hello! Welcome :)

**Comment by NaganathanM** [comment]
Hello everyone!

Hope you're doing well and safe. I'm Naganathan from India. I'm kind of a newbie in tech stack. Previously I was working in a field related to substation protection and automation for five years. I am not new to coding as in college, I had CS subjects and I can understand a program written in C/C++/Python. I have to admit that I'm not an expert level programmer. After a career counselling, I decided to switch to DS/ML. I am planning to pursue masters in DS/ML/AI in Europe. While waiting for results from universities, I'm learning new things in programming and mathematics. I think that's enough for an intro. See ya all around. All please take care and stay safe.

**Reply by rsokl** [reply]
Welcome Naganathan, and thanks for taking the time to introduce yourself! I hope you find this to be a useful resource.

**Comment by TheAmaks** [comment]
Hi, first, I want to express my appreciation to you for making available this resource for free. I'm learning this from a non-programming background and I found PLYMI very useful. 

Meanwhile, I want to draw your attention to one of the links, which is broken. It is found in the section on"Creating Arrays Using Random Sampling":

https://numpy.org/doc/stable/reference/routines.random.html

According to the book, the link is a reference to "many more functions that allows one to "draw from a wide variety of statistical distributions". 

I know it might seem trivial especially given the large number of links you provided in the book. However, I thought that it is an important link to check out and I couldn't locate it on the documentation website: https://numpy.org/doc/stable/reference/routines.array-creation.html

I did, however, found this https://numpy.org/doc/stable/reference/random/index.html, which does not seem to be the same. If you could help to find the right link, it will be highly appreciated.

Thank you

**Reply by rsokl** [reply]
Thank you for bringing this to my attention - I will find an appropriate replacement for this link. I appreciate you taking the time to write this up!

**Reply by rsokl** [reply]
@TheAmaks  I fixed this here: https://github.com/rsokl/Learning_Python/pull/185

It will be reflected in the next HTML update

**Comment by chuakimyou** [comment]
I just wanted to say thanks for putting together this wonderful resource. I too, like others, am coming from a beginner/learner background who wants to use python in my daily career (we make video games). So many of the tools for statistics are just sitting out there and out of pure frustration with the lack of motivation from people around me to use them, I started teaching myself. It's so far been one of the best adventures of my life, I'm loving it!

**Reply by rsokl** [reply]
Fantastic! I am glad that you are finding to be useful. And thanks for taking the time to share the kind words 😄

**Comment by bhav1k** [comment]
Hi, I would like to thank you first for this extremely useful resource.
In the Special Methods page from the Object Oriented Programing module, there's a if statement like below in the __repr__ example:
`if self._needed or self._purchased:`
What condition exactly does this if statement check for? Is it whether self._needed or self._purchased exist or is it if they are empty or not?
Is `bool(self._needed)` and `if self._needed:` the same thing?

**Reply by rsokl** [reply]
Hi @bhav1k so sorry for the very late feedback on this. You are right, that writing `if self._needed:` is equivalent to `if bool(self._needed):`. Python will automatically -- and implicitly -- apply the `bool` function for you.

**Comment by ManosGeramas** [comment]
Hi Ryan,

Just wanted to thank you and everyone who has contributed in the Python Like You Mean It!

This is a great learning source. What I have realized is that if PLYMI says that something is important in Python then I should pay attention.

Again thank you for the quality of the website and your work.

Manos

**Reply by rsokl** [reply]
Hi @manosger,

Thank you for taking the time to leave this feedback. It is heartening and motivating to know that you have found it to be reliable and useful. 

Best regards,
Ryan

**Comment by srec24** [comment]
Hello,

just came to say "Hi" to the community and to thank you for your PLYMI learning.

Srecko

**Comment by UtkarshRai247** [comment]
Hello Ryan,
It is nice to be part of this group! I am looking forward to sharing and learning.
Best Regards,
Utkarsh

**Comment by nikkurthy** [comment]
Hi Ryan, 

I'm an international med graduate and public health student. I'm learning Python to better work with clinical and public health data. It's overwhelming at the moment, but hopefully we can all overcome it together. 

Niharika

**Comment by Alien-Shore** [comment]
Hello PLYMI peoples!

Long-time computer science person here... I had a 40+ yrs career in IT, now recently retired. I have always enjoyed playing with computers, languages, design, ... most everything related. I started doing some light Python coding maybe in 2015 or so, but was still working full time and didn't have much time to really study it. Since retiring, I've been diving deeper, and I found PLYMI while looking for a solid tutorial on OOP techniques in Python. 

As I have been reading [Module 4](https://www.pythonlikeyoumeanit.com/module_4.html) (for about the fifth time!) I was moved to come here to say thank you to @rsokl , @davidmascharka , and the whole team for producing such a useful, high-quality site that's free for the people!  There are so many poorly done 'teach me python' sites out there, and to me PLYMI really stands in a class of it's own! Please do keep up the great work, and THANK YOU!

**Reply by rsokl** [reply]
Thank *you* for taking the time to leave this kind and thoughtful note. It thrills me to know that this resource is helpful for people and that it might even bring some enjoyment. This made my day.
